### ISSUE TITLE
Control deconfiguration...

### ISSUE THREAD

**Issue by IPeluchito** [TASEmulators/fceux#535]
Fceux deconfigures the control when restarting the PC... automatically change JS 0 to JS 255

1. Connect DS4 via bluetooth (Xinput / DS4Windows)
2. Configure Fceux and restart the PC
3. Reconnect DS4, open Fceux and review settings

 - Windows 11 22000.675
 - Latest Dev FCEUX version (non-QT)

**Comment by zeromus** [comment]
I don't believe that. Rebooting the computer can't do anything to fceux's configuration that fceux doesn't do itself when you close and reopen it -- UNLESS the controller is reappearing in a different configuration after the reboot, in which case it's a problem with your drivers.

**Comment by IPeluchito** [comment]
It's not a matter of faith, it's just facts...With emulators like Pcsx2, Dolphin, Duckstation, Bsnes, PPSSPP, NanoBoyAdvance, melonDS, pcengine and others I haven't had any problem.....by the way I have 4 DS4 controllers paired by bluetooth and the only emulator that detects a wrong configuration (JS 255 ¿¿¿Controller # 255???) after reboot is Fceux...

**Comment by zeromus** [comment]
Perhaps all other emulators use xinput and only fceux uses directinput and your driver's directinput implementation is faulty.

**Comment by IPeluchito** [comment]
If you do these steps, Fceux normally detects gamepad 1 as JS 0, but if you then close Fceux and restart the PC, when you reopen Fceux it automatically changes JS 0 to JS 255 and therefore you have to reconfigure the gamepad in Fceux...

**Comment by zeromus** [comment]
I think you have some weird AF driver that gives a different guid every time the system boots or the device connects....
